### ISSUE TITLE
[Tokenizer] Fine Tuning on handwritten text for a new langage

### ISSUE THREAD

**Issue by SkanderBS2024** [Ucas-HaoranWei/GOT-OCR2.0#272]
Hello,

I'm wondering if the tokenizer should be expanded for a new langage (french) for fine-tuning on custom dataset. I have tested the tokenizer and it seems that it does't support french caracters pretty well.

Other question : how many approx number of images is needed to see a decent result (including augmented data).

If the answer is yes, how is the tokenizer expansion achievable please thanks !

**Comment by fuerpy** [comment]
I also want to extend the language capabilities, and I will keep a continuous eye on this issue ：）

**Comment by Ucas-HaoranWei** [comment]
I also don't think it's necessary to extend the tokenizer. Now that the vocabulary is large enough for multiple languages, it's only necessary to train with relevant data.

**Comment by fuerpy** [comment]
OK, it sounds good.！！Today I also checked some non-Latin languages ​​such as Korean and Hindi. This tokenizer can still decode. I prepared a Korean dataset for training. Are you using finetune? I don't quite understand the difference between stage 3 train and fine tune. Which one should I choose?

**Comment by fuerpy** [comment]
I tried to train Korean Lora. I have 70,000 sets of samples. I did not perform any data enhancement. I trained for 1 epoch with a batch size of 8. I used A100 80GB, with  memory occupying about 60GB. The training time was about 3 hours, and the validation set accuracy was 80%. I tested that the horizontal text recognition was OK, but the vertical text recognition was poor. This is the problem that I expect to use a large model better than a small model. I will try some data enhancement methods.

**Comment by fuerpy** [comment]
Btw, does anyone have any method for multi-language testing?

**Comment by zzzzaaa-000** [comment]
方法一：直接添加新词到 Tokenizer
（适合原有 tokenizer 基本可用，但只是缺少一些法语常用词的情况）

收集法语常用词（或你的数据集中高频的法语词/带音标的词）。
用 tokenizer.add_tokens([...]) 增加到词表。
注意：扩展后 embedding 层要同步调整：model.resize_token_embeddings(len(tokenizer))。
示例代码（HuggingFace Transformers）：

python
from transformers import AutoTokenizer, AutoModel

tokenizer = AutoTokenizer.from_pretrained('bert-base-uncased')  # 或你的模型
model = AutoModel.from_pretrained('bert-base-uncased')

new_tokens = ["français", "école", "élève", "éducation"]  # 你的新token
tokenizer.add_tokens(new_tokens)
model.resize_token_embeddings(len(tokenizer))
    具体详细点击我的AI链接    https://code.mayoubang.cn/?_s=git-wz
